### PR TITLE
Mig/implement loki log aggregation prod 11724

### PR DESCRIPTION
**Pull request by mahkoCosmo**

### CODE REVIEW

File: api/kubernetes/deploy_via_helm.sh [hunk 1]
```diff
@@ -70,7 +70,7 @@ echo CHART_PACKAGE_VERSION: $CHART_PACKAGE_VERSION
 echo NAMEPSACE: $NAMESPACE
 echo API_VERSION: $API_VERSION
 
-export ARGO_VERSION="0.16.6"
+export ARGO_VERSION="0.22.15"
```
Comment (jreynard-code) [inline api/kubernetes/deploy_via_helm.sh:73]: As far as I know `deploy_via_helm.sh` is no longer used :thinking: 
@enocquet Can you confirm this please ?

File: scenariorun/src/main/kotlin/com/cosmotech/scenariorun/workflow/argo/ArgoWorkflowService.kt [hunk 2]
```diff
@@ -345,22 +347,28 @@ internal class ArgoWorkflowService(
...
+      workflow.status?.nodes?.forEach { (nodeKey, nodeValue) ->
+        var podName = ""
+        if (workflow.spec.templates?.get(0)?.containerSet != null) {
+          podName = nodeValue.name
+        } else {
+          var name = StringBuilder(nodeValue.id)
+          var toInsert = nodeValue.displayName + "-"
+          name.insert(nodeValue.id.lastIndexOf('-') + 1, toInsert)
+          podName = name.toString()
+        }
```
Comment (jreynard-code) [inline scenariorun/src/main/kotlin/com/cosmotech/scenariorun/workflow/argo/ArgoWorkflowService.kt:362]: Can be extract into internal method

File: scenariorun/src/main/kotlin/com/cosmotech/scenariorun/workflow/argo/WorkflowBuilders.kt [hunk 3]
```diff
@@ -86,7 +86,7 @@ internal fun buildTemplate(
 
   val template =
       IoArgoprojWorkflowV1alpha1Template()
-          .name(scenarioRunContainer.name)
+          .name(scenarioRunContainer.name.lowercase())
```
Comment (jreynard-code) [inline scenariorun/src/main/kotlin/com/cosmotech/scenariorun/workflow/argo/WorkflowBuilders.kt:89]: Why did you put lowercase everywhere?
Is it something needed by solution/project?

**Review by jreynard-code — CHANGES_REQUESTED** [review]
Some remarks.
@enocquet could you please check deployment scripts ?

File: scenariorun/src/main/kotlin/com/cosmotech/scenariorun/workflow/argo/WorkflowBuilders.kt (hunk 3 above)
Comment (mahkoCosmo) [reply scenariorun/src/main/kotlin/com/cosmotech/scenariorun/workflow/argo/WorkflowBuilders.kt:89]: With the Emissary executor, it's a mandatory to have container's name in lowercase

**Review by jreynard-code — COMMENTED** [review]
Seems good to me but @enocquet should review local deployment script <3

**Review by enocquet — APPROVED** [review]

File: api/kubernetes/deploy_via_helm.sh [hunk 4]
```diff
@@ -387,6 +389,42 @@ EOF
...
 helm upgrade --install -n ${NAMESPACE} ${ARGO_RELEASE_NAME} argo/argo-workflows --version ${ARGO_VERSION} --values values-argo.yaml
 
+LOKI_RELEASE_NAME="loki"
+helm repo add grafana https://grafana.github.io/helm-charts
+#Pod errors due to “too many open files”
+#sudo sysctl fs.inotify.max_user_watches=524288
+#sudo sysctl fs.inotify.max_user_instances=512
+#To make the changes persistent, edit the file /etc/sysctl.conf and add these lines
+#fs.inotify.max_user_watches = 524288
+#fs.inotify.max_user_instances = 512
```
Comment (jreynard-code) [inline api/kubernetes/deploy_via_helm.sh:399]: Could you please remove comments?

**Review by jreynard-code — CHANGES_REQUESTED** [review]
Just comments to remove in deploy_via_helm.sh script and it should be ok

**Pull request merged**